File: priv/react.js

var pattern = window.matches.pattern;
var value = pattern({
	'key,o': function(key,o) {
		var props = o['props'];
		return props[key];
	}});
var start = pattern({
	'': function() {
		var user = React.createClass({render: function() {
		return React.DOM.h1(null,value('email',this));;
	}});;
		var commentlist = React.createClass({props: {data:[]},
render: function() {
		var users = value('data',this).map(function(item) {
		return user({props: item});;
	});
;
		return users;
	}});;
		return commentlist;
	}});
start();

File: src/react.erl
-module(react).
-compile({parse_transform, shen}).
-compile(export_all).

-output("priv").
-js([start/0,value/2]).

-record(react,{props=[],state,name,init,render,willMount=fun(X)->X end}).

value(Key,O) -> Props = O:at("props"), Props:at(Key).

start() -> 
    User = #react{
        render = fun(This) -> value(email,This) end
            },

    CommentList = #react{
        props = [{data,[]}],
        render = fun(This) -> 
            Users = lists:map(fun(Item) ->
                User#react{props=Item} end, value(data,This)),
            Users
        end
    },

    CommentList.
